src/agentconfig.rs

```diff
@@ -19,23 +19,69 @@ pub enum AgentConfigError {
 
 #[derive(Clone)]
 pub struct AgentConfig {
+    pub hostname: String,
     pub parcel_directory: PathBuf,
     pub config_directory: PathBuf,
     pub log_directory: PathBuf,
+    pub bootstrap_file: PathBuf,
+    pub data_directory: PathBuf,
+    pub plugin_directory: PathBuf,
     pub server_ip_address: IpAddr,
+    pub server_port: u16,
     pub server_cert_file: Option<PathBuf>,
     pub server_key_file: Option<PathBuf>,
     pub tags: HashMap<String, String>,
 }
 
 impl AgentConfig {
+    pub const HOSTNAME: ConfigOption = ConfigOption {
+        name: "hostname",
+        default: None,
+        required: false,
+        takes_argument: true,
+        help:
+            "The hostname to register the node under in Kubernetes - defaults to system hostname.",
+        documentation: include_str!("config_documentation/hostname.adoc"),
+        list: false,
+    };
+
+    pub const DATA_DIR: ConfigOption = ConfigOption {
+        name: "data-directory",
+        default: Some("/etc/stackable/agent"),
+        required: false,
+        takes_argument: true,
+        help: "The directory where the stackable agent should keep its working data.",
+        documentation: include_str!("config_documentation/data_directory.adoc"),
+        list: false,
+    };
+
+    pub const PLUGIN_DIR: ConfigOption = ConfigOption {
+        name: "plugin-directory",
+        default: Some("/etc/stackable/agent/plugins"),
+        required: false,
+        takes_argument: true,
+        help: "The directory to observe for new sockets to appear which can be used to communicate with CSI plugins.",
+        documentation: include_str!("config_documentation/plugin_directory.adoc"),
+        list: false,
+    };
+
+    pub const BOOTSTRAP_FILE: ConfigOption = ConfigOption {
+        name: "bootstrap-file",
+        default: Some("/etc/kubernetes/bootstrap-kubelet.conf"),
+        required: false,
+        takes_argument: true,
+        help: "The bootstrap file to use in case Kubernetes bootstraping is used to add the agent.",
```

Comment:
What's Kubernetes bootstrapping? Do we support this? I assume it comes from the Krustlet?

Reply:
That is the process of connecting with a temporary token and requesting TLS certificates for the "real" connection.

https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/

I have not tested this yet, but there  is code in the Krustlet that to me looks like performing that function. Getting that to work is on my todo list right after the systemd stuff.